openscap_report/scap_results_parser/scap_results_parser.py

```diff
@@ -107,4 +107,11 @@ def parse_report(self):
         self._debug_show_rules(rules)
         report.rules = rules
         report.groups = groups
+
+        if len(report.profile_info.selected_rules_ids) > 0:
+            report.profile_info.selected_rules_ids.extend(rule_parser.to_select_rule_ids)
```

Comment:
That should be a `ProfileInfo` mutate method, like `extend_selected_rules_ids`. Along with the logic how/when to do that `if len(report.profile_info.selected_rules_ids) > 0:`.

Why do you update it only when it is not empty by the way?

Reply:
There are two approaches how to determine if the rule is selected. 
- First is to get information from elements `<xccdf-1.2:select ...>`. A list of `selected_rules_ids` represents these elements. 
- The second is to get information from the result of the rule. 

The second approach is used if select elements aren't present in the ARF report. That is why the list is not extended when the list is empty.